src/segment_plastic.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type plastic struct {
+	props properties
+	env   environmentInfo
+
+	Moved    int
+	Deleted  int
+	Added    int
+	Modified int
+	Unmerged int
+	Changed  bool
+
+	Behind   bool
+	Selector string
+
+	plasticWorkspaceFolder string // .plastic working folder
+
+	cmCommand string
+}
+
+const (
+	// FullBranchPath displays the full path of a branch
+	FullBranchPath Property = "full_branch_path"
+)
+
+func (g *plastic) enabled() bool {
+	if !g.env.hasCommand(g.getCmCommand()) {
+		return false
+	}
+	wkdir, err := g.env.hasParentFilePath(".plastic")
+	if err != nil {
+		return false
+	}
+	if g.shouldIgnoreRootRepository(wkdir.parentFolder) {
+		return false
+	}
+
+	if wkdir.isDir {
+		g.plasticWorkspaceFolder = wkdir.parentFolder
+		return true
+	}
+
+	return false
+}
+
+func (g *plastic) shouldIgnoreRootRepository(rootDir string) bool {
+	value, ok := g.props[ExcludeFolders]
+	if !ok {
+		return false
+	}
+	excludedFolders := parseStringArray(value)
+	return dirMatchesOneOf(g.env, rootDir, excludedFolders)
+}
+
+func (g *plastic) string() string {
+	statusColorsEnabled := g.props.getBool(StatusColorsEnabled, false)
+	displayStatus := g.props.getOneOfBool(FetchStatus, DisplayStatus, false)
+
+	g.Selector = g.getSelector()
+	if displayStatus || statusColorsEnabled {
+		g.getPlasticStatus()
+	}
+
+	// use template if available
+	segmentTemplate := g.props.getString(SegmentTemplate, "")
+	if len(segmentTemplate) > 0 {
+		return g.templateString(segmentTemplate)
+	}
+
+	// default: only selector is returned
+	return g.Selector
+}
+
+func (g *plastic) templateString(segmentTemplate string) string {
+	template := &textTemplate{
+		Template: segmentTemplate,
+		Context:  g,
+		Env:      g.env,
+	}
+	text, err := template.render()
+	if err != nil {
+		return err.Error()
+	}
+	return text
+}
+
+func (g *plastic) Status() string {
```

Comment:
This is still based on the previous version of the git segment, it would be interesting to see if the `gitStatus` struct could be reused (and renamed obviously in that case).

Reply:
I agree. It could be renamed to `scmStatus`. What do you think?
One difference is that I counted `moved` files separatly and in `git` they are counted as `modified`.